lib/rubocop/cop/style/redundant_begin.rb

```diff
@@ -131,6 +132,12 @@ def empty_begin?(node)
           node.children.empty?
         end
 
+        def oneliner_with_semicolon?(node)
```

Comment:
I think it's better to count the number of child nodes in the begin - it's one-line and has 1+ children they must be separated by semicolons, right?

Reply:
Right! And it should have been accepted if `begin` block has multiple statements, whether one-liner or multiple-lines. I also added several test cases and modified this implementation based on it.